artemis-protocols/artemis-proton-plug/src/main/java/org/proton/plug/util/DeliveryUtil.java

```diff
@@ -30,6 +30,9 @@ public static int readDelivery(Receiver receiver, ByteBuf buffer) {
       while ((count = receiver.recv(buffer.array(), buffer.arrayOffset() + buffer.writerIndex(), buffer.writableBytes())) > 0) {
          // Increment the writer index by the number of bytes written into it while calling recv.
          buffer.writerIndex(buffer.writerIndex() + count);
+         if (buffer.writableBytes() <= 0) {
+            buffer.ensureWritable(count);
```

Comment:
You are right Clebert, we don't need it. The netty api says if there are enough writeable bytes this call won't do anything. I'll remove the 'if' checking. Thanks